browser/brave_rewards/DEPS

```diff
@@ -1,5 +1,5 @@
 include_rules = [
-  "+brave/components/brave_rewards/core",
+  "+brave/components/brave_rewards/core/internal",
```

Comment:
This seems odd, `internal` code shouldn't be accessible outside of the component. This would also allow subdirectories inside `core/internal`. Maybe this should have a different directory name or maybe whatever files you need in `browser/brave_rewards` should go somewhere else?

Reply:
Right, this is why I suggested moving things that are shared out of internal, but it seems like this PR is moving a lot of things into internal? In any case I think it's best to limit this PR to fixing of layered component directory issues and follow-up with other changes to keep the size/scope of this PR down